src/app/components/Embeds/UnsupportedEmbed/index.test.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { render } from '../../react-testing-library-with-providers';
+import { ServiceContextProvider } from '../../../contexts/ServiceContext';
+import {
+  RequestContext,
+  RequestContextProps,
+} from '../../../contexts/RequestContext';
+import { ARTICLE_PAGE } from '../../../routes/utils/pageTypes';
+import { Services } from '../../../models/types/global';
+import sampleTelescopeProps from './fixtures';
+import UnsupportedEmbed from '.';
+import { UnsupportedEmbedProps } from '../types';
+
+const Component = ({
+  props,
+  isAmp,
+  service = 'news',
+}: {
+  props: UnsupportedEmbedProps;
+  isAmp: boolean;
+  service?: Services;
+}) => (
+  <RequestContext.Provider
+    value={
+      {
+        id: 'c0000000000o',
+        isAmp,
+        isApp: false,
+        pageType: ARTICLE_PAGE,
+        pathname: '/pathname',
+        service,
+        statusCode: 200,
+        canonicalLink: 'canonical_link',
+      } as unknown as RequestContextProps
+    }
+  >
+    <ServiceContextProvider service={service}>
+      <UnsupportedEmbed {...props} />
+    </ServiceContextProvider>
+  </RequestContext.Provider>
+);
+
+describe('Unsupported Embed', () => {
+  describe('Canonical', () => {
+    it('Returns null', () => {
+      const { container } = render(
+        <Component props={sampleTelescopeProps} isAmp={false} />,
+      );
+      expect(container).toBeEmptyDOMElement();
+    });
+  });
+
+  describe('AMP', () => {
+    it('Renders Embed Error message with a link to the canonical page', () => {
+      const { container, getByText } = render(
+        <Component props={sampleTelescopeProps} isAmp />,
```

Comment:
The `render` from `react-testing-library-with-providers` can take a second argument of `options`. The function should provide all the contexts needed to render a page, so you should be able to remove the `<Component />` wrapper and just use the `<UnsupportedEmbed />` component directly with `options` passed in like:
```
const { container, getByText } = render(
        <UnsupportedEmbed {...sampleTelescopeProps} />,
        {
          isAmp: true,
          isApp: false,
          id: 'c0000000000o',
          pageType: ARTICLE_PAGE,
          pathname: '/pathname',
          statusCode: 200,
        },
      );
```

Reply:
That looks a lot better! I've used the /pathname to check the canonicalLink value, since I couldn't pass this in as an option.